presto-native-execution/presto_cpp/main/connectors/arrow_flight/tests/ArrowFlightConnectorTlsTest.cpp

```diff
@@ -0,0 +1,111 @@
+namespace exec = facebook::velox::exec;
+namespace connector = facebook::velox::connector;
+namespace flight = arrow::flight;
+
+using namespace facebook::presto::connector::afc;
+using namespace facebook::presto::connector::afc::test;
+using exec::test::AssertQueryBuilder;
+using exec::test::OperatorTestBase;
+
+class FlightConnectorTlsTestBase : public FlightWithServerTestBase {
+ protected:
+  explicit FlightConnectorTlsTestBase(
+      std::shared_ptr<velox::config::ConfigBase> config)
+      : FlightWithServerTestBase(
+            std::move(config),
+            createFlightServerOptions(
+                true, /* isSecure */
+                "./data/server.crt",
+                "./data/server.key")) {}
+
+  void executeTest(bool isPositiveTest = true) {
+    updateTable(
+        "sample-data",
+        makeArrowTable(
+            {"id"},
+            {makeNumericArray<arrow::Int64Type>(
+                {1, 12, 2, std::numeric_limits<int64_t>::max()})}));
+
+    auto idVec = makeFlatVector<int64_t>(
+        {1, 12, 2, std::numeric_limits<int64_t>::max()});
+
+    auto plan = FlightPlanBuilder()
+                    .flightTableScan(velox::ROW({"id"}, {velox::BIGINT()}))
+                    .planNode();
+
+    auto loc = std::vector<std::string>{"grpc+tls://localhost:5000"};
+    if (isPositiveTest) {
+      AssertQueryBuilder(plan)
+          .splits(makeSplits({"sample-data"}, loc))
+          .assertResults(makeRowVector({idVec}));
+    } else {
+      VELOX_ASSERT_THROW(
+          AssertQueryBuilder(plan)
+              .splits(makeSplits({"sample-data"}, loc))
+              .assertResults(makeRowVector({idVec})),
+          "Server replied with error");
+    }
+  }
+};
+
+class FlightConnectorTlsTest : public FlightConnectorTlsTestBase {
+ protected:
+  explicit FlightConnectorTlsTest()
+      : FlightConnectorTlsTestBase(std::make_shared<velox::config::ConfigBase>(
+            std::unordered_map<std::string, std::string>{
+                {FlightConfig::kServerVerify, "true"},
+                {FlightConfig::kServerSslCertificate, "./data/ca.crt"}})) {}
+};
+
+TEST_F(FlightConnectorTlsTest, TlsTest) {
+  executeTest();
+}
+
+class FlightConnectorTlsNoCertValidationTest
+    : public FlightConnectorTlsTestBase {
+ protected:
+  explicit FlightConnectorTlsNoCertValidationTest()
+      : FlightConnectorTlsTestBase(std::make_shared<velox::config::ConfigBase>(
+            std::unordered_map<std::string, std::string>{
+                {FlightConfig::kServerVerify, "false"}})) {}
+};
+
+TEST_F(FlightConnectorTlsNoCertValidationTest, TlsNoCertValidationTest) {
+  executeTest();
+}
+
+class FlightConnectorTlsNoCertTest : public FlightConnectorTlsTestBase {
+ protected:
+  FlightConnectorTlsNoCertTest()
+      : FlightConnectorTlsTestBase(std::make_shared<velox::config::ConfigBase>(
+            std::unordered_map<std::string, std::string>{
+                {FlightConfig::kServerVerify, "true"}})) {}
```

Comment:
This isn't necessarily a bad config. We could be dealing with servers which use certificates signed by trusted CAs, in which case providing additional certificates is unnecessary.

In this particular case, the reason this is a purely negative test is because our test server is set up with a self-signed certificate, and it's impractical to acquire a certificate signed by a trusted CA just for this purpose.